### PR TITLE
#184 Referencing sub-resources + composite identifiers

### PR DESCRIPTION
**Pull request by maxim-tschumak**
#184 Adding some guidelines about how to reference sub-resources and how to design composite identifiers, as discussed in the API Guild JF.

Any feedback is welcome!

### CODE REVIEW

File: resources/Resources.md [hunk 1]
```diff
@@ -28,14 +28,25 @@ API resources represent elements of the application’s domain model. Using doma
...
 ## {{ book.must }} Identify resources and Sub-Resources via Path Segments
 
+Some API resources may contain or reference sub-resources. Embedded sub-resources, which are not top-level resources,
+are parts of a higher-level resource and cannot be used outside of its scope. Sub-resources should be referenced
+by their name and identifier in the path segments.
+
+Composite identifiers must not contain “/” as a separator. In order to improve the consumer experience, you should
+aim for intuitively understandable URLs, where each sub-path is a valid reference to a resource or a set of resources.
+For example, if “/customers/12ev123bv12v/addresses/DE\_100100101” is a valid path of your API, then
+“/customers/12ev123bv12v/addresses”, “/customers/12ev123bv12v” and “/customers” must be valid as well.
```
Comment (codekultur) [inline resources/Resources.md]: Really like the PR! Here I think we should say "must be valid as well in principle" - as the API does not really need to support operations on these endpoints (e.g. if there are no use cases).

Comment (maxim-tschumak) [inline resources/Resources.md]: done, thanks for the comment!

**Comment by ePaul** [comment]
:+1:

**Comment by ePaul** [comment]
Possibly add a changelog entry?
(Though this might get conflicts with other PRs.)

**Comment by fmueller** [comment]
👍 

@ePaul Can you re-thumb after @maxim-tschumak added changelog entry? ;)

**Comment by fmueller** [comment]
👍

**Comment by ePaul** [comment]
:+1:

**Pull request merged**